Source/Scene/PointPrimitive.js

```diff
@@ -0,0 +1,488 @@
+/*global define*/
+define([
+        '../Core/Cartesian2',
+        '../Core/Cartesian3',
+        '../Core/Cartesian4',
+        '../Core/Color',
+        '../Core/createGuid',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/Matrix4',
+        '../Core/NearFarScalar',
+        './SceneMode',
+        './SceneTransforms'
+    ], function(
+        Cartesian2,
+        Cartesian3,
+        Cartesian4,
+        Color,
+        createGuid,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        Matrix4,
+        NearFarScalar,
+        SceneMode,
+        SceneTransforms) {
+    "use strict";
+
+    /**
+     * A glPoint positioned in the 3D scene, that is created
+     * and rendered using a {@link PointPrimitiveCollection}.  A pointPrimitive is created and its initial
+     * properties are set by calling {@link PointPrimitiveCollection#add}.
+     *
+     * @alias PointPrimitive
+     *
+     * @performance Reading a property, e.g., {@link PointPrimitive#show}, is constant time.
+     * Assigning to a property is constant time but results in
+     * CPU to GPU traffic when {@link PointPrimitiveCollection#update} is called.  The per-pointPrimitive traffic is
+     * the same regardless of how many properties were updated.  If most pointPrimitives in a collection need to be
+     * updated, it may be more efficient to clear the collection with {@link PointPrimitiveCollection#removeAll}
+     * and add new pointPrimitives instead of modifying each one.
+     *
+     * @exception {DeveloperError} scaleByDistance.far must be greater than scaleByDistance.near
+     * @exception {DeveloperError} translucencyByDistance.far must be greater than translucencyByDistance.near
+     *
+     * @see PointPrimitiveCollection
+     * @see PointPrimitiveCollection#add
+     * @see Label
+     *
+     * @internalConstructor
+     *
+     * @demo {@link http://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=PointPrimitives.html|Cesium Sandcastle PointPrimitive Demo}
```

Comment:
I don't think this demo exists.  I would like a trivial example though in the `development` tab to for @bagnell, myself, and others who are doing the low-level work.